fastlane_bot/tests/deterministic/dtest_token.py

```diff
@@ -0,0 +1,79 @@
+"""
+Token class to store token address and contract object for interacting with the token on the blockchain.
+
+(c) Copyright Bprotocol foundation 2024.
+Licensed under MIT License.
+"""
+from dataclasses import dataclass
+
+from eth_typing import Address
+from web3 import Web3
+from web3.contract import Contract
+
+from fastlane_bot.tests.deterministic.dtest_constants import ETH_ADDRESS
+
+
+@dataclass
+class TestToken:
+    """
+    A class to represent a token on the blockchain.
+
+    Attributes:
+        address: str or Address
+    """
+
+    address: str or Address  # Address after __post_init__, str before
+
+    def __post_init__(self):
+        self.address = Web3.to_checksum_address(self.address)
+        self._contract = None
+
+    @property
+    def contract(self):
+        return self._contract
+
+    @contract.setter
+    def contract(self, contract: Contract):
+        self._contract = contract
```

Comment:
I think that a property is an overkill in here.
Especially since there is no actual logic in here other than returning this member variable as is.
Also, it can possibly go into the `__post_init__` function as an input argument, so it isn't required to initialize that property in a separate line.